### PR TITLE
Finding the Nth Element in an Array

### PR DESCRIPTION
**Pull request by Sohammhatre10**
# New Nada Example

- [x] I have read the [contributing docs](/NillionNetwork/nada-by-example/blob/main/CONTRIBUTING.md)
- [x] This is not a duplicate of any [existing pull request](https://github.com/NillionNetwork/nada-by-example/pulls)

## Description

This example introduces a program that helps users find the Nth largest element in a list or array of `SecretIntegers`. The program solves a common problem of selecting a specific element by size from a given dataset, while maintaining security by handling secret integers. The example is further extended to show how Nada can handle multi-dimensional arrays, adding flexibility and depth for more complex use cases.

### Changes:
- Added `nth_element_of_an_array.py` to the `src/` directory.
- Added `nth_element_of_an_array_test.yaml` to the `tests/` directory for testing various edge cases and array dimensions.
- Updated `nada-project.toml` to include this new program.

## Use Case

The target use cases for this example include:
- Retrieving the Nth largest element from a dataset, useful in sorting algorithms, competitive programming, and data analysis.
- Securely working with `SecretIntegers` to ensure privacy during computations.

## Related Issues
~~This pull request addresses issue https://github.com/NillionNetwork/nada-by-example/issues/17, which requested a new example for selecting the Nth element from an array.~~

## Optional: add your information for a Nada by Example Contributor POAP

Your Twitter: 
Your Ethereum address:

### CODE REVIEW

File: src/nth_element_of_an_array.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    
+    party_alice = Party(name="Party_Alice")
+    parties = na.parties(size)
+    n = SecretInteger(Input(name="n", party=party_alice))
+
+    
+    list = [SecretInteger(Input(name=f"num_{i}", party=parties[i])) for i in range(size)]
+    
+    
+    total_size = Integer(size)
```
Comment (jimouris) [inline src/nth_element_of_an_array.py:16]: why have both `size` and `total_size`?

File: src/nth_element_of_an_array.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    element_3 = Integer(1)
```
Comment (jimouris) [inline src/nth_element_of_an_array.py:17]: what is `element_3`? remove this?

File: src/nth_element_of_an_array.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
+from nada_dsl import *
+import nada_numpy as na
+
+size = 5
+
+def nada_main():
+
```
Comment (jimouris) [inline src/nth_element_of_an_array.py:7]: Can you add some comments here on what this program does? Something like:
```
'''
Given a list and a secret index n, return the element in that index without revealing n.
''' 
```

File: src/nth_element_of_an_array.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    
+    
+    total_size = Integer(size)
+    element_3 = Integer(1)
+
+    element = Integer(0)
+    index = total_size - n
+    for i in range(size):
+        number = Integer(i)
+        element += (number == index).if_else(list[i], Integer(0))
```
Comment (jimouris) [inline src/nth_element_of_an_array.py:23]:
```suggestion
        element += (Integer(i) == index).if_else(list[i], Integer(0))
```

File: src/nth_element_of_an_array.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+    
+    list = [SecretInteger(Input(name=f"num_{i}", party=parties[i])) for i in range(size)]
+    
+    
+    total_size = Integer(size)
+    element_3 = Integer(1)
+
+    element = Integer(0)
+    index = total_size - n
```
Comment (jimouris) [inline src/nth_element_of_an_array.py:20]: Shouldn't `index` be just `n`?

**Review by jimouris — CHANGES_REQUESTED** [review]

**Review by jimouris — APPROVED** [review]

**Review by oceans404 — APPROVED** [review]
